src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -96,9 +96,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_handle_short_am(MPIDI_OFI_am_header_t * m
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_HANDLE_SHORT_AM);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_HANDLE_SHORT_AM);
 
-    /* note: setting is_local, is_async, req to 0, 0, NULL */
+    int attr = 0;               /* is_local = 0, is_async = 0 */
     MPIDIG_global.target_msg_cbs[msg_hdr->handler_id] (msg_hdr->handler_id, am_hdr,
-                                                       p_data, msg_hdr->payload_sz, 0, 0, NULL);
+                                                       p_data, msg_hdr->payload_sz, attr, NULL);
```

Comment:
There are other occurrences below.

Reply:
The callback is an opaque function to the compiler, it is not able to optimize away the parameters anyway.

Reply:
Yes. What I forgot to mention is passing constant is cheaper because it does not need to read the variable. An initialized variable is not a replacement for a constant.

Reply:
The variable does not exist after the compiler optimization.